exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/decoder/TcpSession.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.pcap.decoder;
+
+import org.joda.time.Instant;
+import org.joda.time.Period;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static org.apache.drill.exec.store.pcap.PcapFormatUtils.parseBytesToASCII;
+
+/**
+ * This class is the representation of a TCP session.
+ */
+public class TcpSession {
+
+  private ArrayList<Packet> packetsFromSender;
+  private ArrayList<Packet> packetsFromReceiver;
+
+  private long startTime;
+  private long endTime;
+  private int packetCount;
+  private InetAddress srcIP;
+  private InetAddress dstIP;
+  private int srcPort;
+  private int dstPort;
+  private String srcMac;
+  private String dstMac;
+  private long sessionID;
+  private TcpHandshake handshake;
+  private long synTime;
+  private long ackTime;
+  private long connectTime;
+  private byte[] sentData;
+  private byte[] receivedData;
+  private int sentDataSize;
+  private int receivedDataSize;
+  private boolean hasCorruptedData = false;
+
+
+  private static final Logger logger = LoggerFactory.getLogger(TcpSession.class);
+
+  public TcpSession (long sessionID) {
+    packetsFromSender = new ArrayList<>();
```

Comment:
Please don't forget that anything that is initialized in the constructor and is not going to be changed should be marked as final.

Reply:
Fixed